VikingQuiz.Api/Utilities/AzureBlobService.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Microsoft.WindowsAzure.Storage.Shared.Protocol;
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.IO;
+using System.Linq;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using VikingQuiz.Api.Models;
+
+namespace VikingQuiz.Api.Utilities
+{
+    public class AzureBlobService
+    { // should be changed to a SINGLETON if posibile
+        private CloudStorageAccount account;
+        private CloudBlobClient blobClient;
+        private ServiceProperties serviceProperties;
+        private CloudBlobContainer userContainer;
+        public Uri urlPath;
+
+        public async Task<bool> InitializeBlob()
+        {
+            userContainer = blobClient.GetContainerReference("users");
+            if (!await userContainer.CreateIfNotExistsAsync())
+            {
+                return false;
+            }
+            await userContainer.CreateIfNotExistsAsync();
+            BlobContainerPermissions permissions = new BlobContainerPermissions
+            {
+                PublicAccess = BlobContainerPublicAccessType.Blob
+            };
+            await userContainer.SetPermissionsAsync(permissions);
+            serviceProperties = await SetServiceProperties();
+            return true;
+
+        }
+
+        private async Task<ServiceProperties> SetServiceProperties()
+        {
+            var serviceProperties = await blobClient.GetServicePropertiesAsync();
+            serviceProperties.Cors.CorsRules.Clear();
+            serviceProperties.Cors.CorsRules.Add(new CorsRule()
+            {
+                AllowedHeaders = { "*" },
+                AllowedMethods = CorsHttpMethods.Get | CorsHttpMethods.Head | CorsHttpMethods.Post | CorsHttpMethods.Put | CorsHttpMethods.Delete,
+                AllowedOrigins = { "*" },
+                ExposedHeaders = { "*" },
+                MaxAgeInSeconds = 1800
+            });
+
+            await blobClient.SetServicePropertiesAsync(serviceProperties);
+
+            return serviceProperties;
+        }
+
+
+        public async Task<string> UploadPhoto(IFormFile file)
+        {
+            var filePath = Path.GetTempFileName();
+            using (var stream = new FileStream(filePath, FileMode.Create)) {
+                await file.CopyToAsync(stream);
+            }
+
+            string contentType = file.ContentType;
+            var extension = contentType == "image/png" ? ".png" :
```

Comment:
Instead of this ugly sintax, you can use either a Dictionary for contentType or some other method which would return you the right type.